src/test/java/com/softserve/academy/spaced/repetition/controller/CourseRatingControllerTest.java

```diff
@@ -0,0 +1,104 @@
+package com.softserve.academy.spaced.repetition.controller;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import com.softserve.academy.spaced.repetition.domain.Course;
+import com.softserve.academy.spaced.repetition.domain.CourseRating;
+import com.softserve.academy.spaced.repetition.exceptions.NotAuthorisedUserException;
+import com.softserve.academy.spaced.repetition.service.CourseRatingService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CourseRatingControllerTest {
+
+    private MockMvc mockMvc;
+
+    @InjectMocks
+    private CourseRatingController courseRatingController;
+
+    @Mock
+    private CourseRatingService courseRatingService;
+
+    @Before
+    public void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(courseRatingController)
+                .setControllerAdvice(new ExceptionHandlerController())
+                .alwaysDo(print())
+                .build();
+    }
+
+    @Test
+    public void getCourseRatingById() throws Exception {
+        when(courseRatingService.getCourseRatingById(eq(5L))).thenReturn(createCourseRating());
+        mockMvc.perform(get("/api/rate/course/{id}", 5L)
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("{\"rating\":3,\"accountEmail\":\"email@email\",\"courseId\":5,\"links\":[{\"rel\":\"self\",\"href\":\"http://localhost/api/rate/course/77\"}]}"));
```

Comment:
Instead of "links" should be "_links", isn't it?

Reply:
Yes, I agree.

Reply:
Oh, and, besides, you have forgotten to add the port here.

Reply:
Doesn't matter the port - for the test it is sufficient.

The problem is that course has id=5, but link is "api/rate/course/77", I suspect this is wrong.

Reply:
@antonbakalets, but this method returns the rating by its id, not by id of course.